Add vision and guiding principles document (#7)
Hubs-Foundation/policies-procedures-guidelines-public: vision-and-guiding-principles.md

# Vision and Guiding Principles

## Vision

> Our vision is simple: to build an open, decentralized, high quality, unopinionated, web-based metaverse that puts freedom of expression, privacy, community, and integrity first.

Reviewer: Crypto would presumably belong in an add-on. There's a lot of initiatives besides crypto under the umbrella of "decentralized".  "Web3" is a marketing term more associated with crypto.

I think it's best not to mention crypto at all.

> Our vision is simple: to build an open, decentralized, high quality, unopinionated, web-based metaverse that puts freedom of expression, privacy, community, and integrity first.

Reviewer 2: Both valid points.  I tend to lean towards Doug's conclusion, however: best not to mention (or not to mention HERE).

> Our vision is simple: to build an open, decentralized, high quality, unopinionated, web-based metaverse that puts freedom of expression, privacy, community, and integrity first.

Author: @nurse-the-code 
You do have a good point.  There is already Decentraland and DAOs (Decentralized Autonomous Organizations) and a quick web search seems to show a lot of literature referring to decentralized metaverses as blockchain powered.  Although decentralized is used a lot in non crypto/blockchain contexts (e.g. Git is decentralized and no one would think it has anything to do with crypto), in the metaverse context it might be confusing.  However, hopefully by the time people are reading this document (they'll probably have been through the hubsfoundation.org home page at least) they should have a bit of a sense of what Hubs is and should know that it doesn't have anything to do with crypto (at least right now.  Maybe someone will create an add-on for it someday).

Another thing to consider is that adding that we aren't crypto-based to *this* document would violate our "unopinionated" principle.  So, I think I agree that we shouldn't mention it here, but we may want to add something to the hubsfoundation.org home page clarifying that Hubs currently doesn't have anything to do with crypto.

> Our vision is simple: to build an open, decentralized, high quality, unopinionated, web-based metaverse that puts freedom of expression, privacy, community, and integrity first.

Author: @DougReeder I don't know your thoughts on mentioning crypto outside this document (your reply suggests to me that you might be against it), but I was looking at the Hubs Foundation's homepage and I think it could potentially use some revision to better describe what Hubs is (to me the homepage kinda suggests it's a single instance, although the get started page does briefly mention multiple instances), and I think a description of what we mean when we say decentralized could fit well there.

## Guiding Principles

Our guiding principles are the firm foundation that we stand on to bring our vision into reality.

### Unopinionated Freedom of Expression

We believe our job is to empower people to create what they want; that we should create technology and tools that are flexible and that don't constrain people to a single art style or way of doing things, but rather enables them to have/create the experience they want.  This also means that we support minority features/use cases.  So whether 100% of people will need it or only 1%, if we can reasonably provide it, or a way to support them creating it, that doesn't inhibit the rest of our users, we will endeavour to do so.

### Quality

> We want to build a world class metaverse that is solid and dependable.  To do that, we impose strict quality controls on all contributions (both internal and external) and work together to make sure each feature is carefully designed from the beginning so that it fits into our vision.

Author: In addition to @DougReeder's suggestion, what about rewriting the second sentence like so?  Modifications are bolded.

"To do that, we **do the work that is needed (even though it might not be glamorous), take the time needed to do a good job,** impose strict quality controls on all contributions (both internal and external), and work together to make sure each **change** is carefully designed from the beginning so that it fits into our vision."

### Openness and Transparency

We believe that by making everything open and transparent, at all levels, we empower people, encourage collaboration, experimentation, participation, and that the whole project benefits as a result.

### Privacy

We believe that privacy is important, that you should have control of what you share, that you shouldn't be tracked without your consent, and that you should be able to feel safe and confident with the the tools and services you use and the metaverse you create.

> ### Open Source
> 
> As an open source project ourselves, we value and prioritize open source and the freedom and control it gives us to take full command of our own experiences.  While we don't object in general to proprietary software, we will either seek open source solutions first, or at least try to make sure there is an open source option.

Author: The following is a rewrite to include a paraphrase/expansion of the four essential freedoms of free/libre software, customized for our community/project and a clarification on where we stand as kind of a hybrid between the two philosophies of free/libre software and open source software.

References:
https://www.gnu.org/philosophy/free-sw.html
https://www.gnu.org/philosophy/open-source-misses-the-point.html
https://opensource.org/osd
```suggestion
### Freedom and Open Source 

We as a project believe we all should have the freedom to:
* use the software we create as we wish, for any purpose (including in combination with proprietary software)
* learn how the software (and larger project) works
* change how our software works to suit our needs
* share the software, any modifications of it, and knowledge about it with others
* offer contributions back to the project and get involved
* not have to worry that it stays accessible and open to all for the common good (most of our stuff is under the [MPL-2.0 license, which is a weak copyleft license](https://web.archive.org/web/20250513025257/https://milvus.io/ai-quick-reference/what-are-the-restrictions-of-the-mozilla-public-license-mpl))

In addition, as a free/libre open source software project ourselves, we value and prioritize free software  and open source and the freedom and control it gives us to take full command of our own experiences.  While we don't object to people using proprietary software, we will either seek free/libre or open source solutions first, or at least try to make sure there is a free/libre or open source option.

See https://www.gnu.org/philosophy, https://www.fsf.org/, and https://opensource.org/ for more information on free/libre software and open source software (we're kinda a hybrid).
```

### Inclusion and Diversity and Mutual Respect

> We welcome everyone from everywhere and believe that by cultivating a culture of inclusion and sharing that is based on mutual respect that we will enrich each other and make a better world.

Author: I initially had something similar to this as part of the quality principle (to indicate that the high quality required wasn't a way to exclude people), but ultimately decided it wasn't necessary for that principle and removed it.  However after a discussion with @hobbs-Hobbler, I think introducing this sentiment to the inclusion principle would be good (and it also reinforces our culture of helping people/sharing knowledge).
```suggestion
We welcome everyone from everywhere and believe that by cultivating a culture of inclusion and sharing that is based on mutual respect that we will enrich each other and make a better world.  As a part of this, we help people grow by sharing knowledge, support, ideas, assets, and working with them to ensure that they are able to participate and bring their own unique voice and ideas to the project/community.
```

Reviewer 2: @hobbs-Hobbler  come back and review this.

### Integrity

In a world where too often it seems that money is the only important thing, we value doing what's right and sticking by our principles.  That means that in everything we do, from the products we create to the services we use and/or recommend, we will do our best to make sure they align with our principles.  This isn't always easy (or sometimes even fully possible), so we must all do our part to keep us on track and speak up when needed.  We want to create something that we can all feel good about and proud of, without reservation.

### Community and Culture

The community and its culture of giving is at the heart of our project.  Each of us comes from the community and so we all decide the future of Hubs together.  All of us play a part in defining our culture by the choices we make, the little things that we do, and the time we put in.  This is what makes it worth our time to keep participating, and we believe in preserving and celebrating the community and all the little (and big) things that make it special.  Hubs will survive as long as there are devoted people who care about it.

> ### Fun and Enjoyment and Wonder

Reviewer 3: ```suggestion
### Fun, Enjoyment and Wonder
```

> We believe that fun and enjoyment and wonder are important.and coexist with and enhance the other high bars that we set for the project.  Let's face it, if we aren't evoking these emotions in people then we're doing something wrong because what could be more awesome or inspiring than being able to create and shape a shared metaverse, from the three dimensional constructs to the underlying code!  Hubs presents us with infinite possibilities and we believe it's our job to embrace it and make sure this is highlighted and never lost.

Reviewer 2: I think it's funny that most of us here LOVE the Oxford comma and as soon as it goes missing, we're all like (see picture) so much that @Exairnous has to explain that he's being "fun". 
![when an oxford comma is missed](https://github.com/user-attachments/assets/2666549e-0fe2-496d-b88d-cede3bf0288f)

> We believe that fun and enjoyment and wonder are important.and coexist with and enhance the other high bars that we set for the project.  Let's face it, if we aren't evoking these emotions in people then we're doing something wrong because what could be more awesome or inspiring than being able to create and shape a shared metaverse, from the three dimensional constructs to the underlying code!  Hubs presents us with infinite possibilities and we believe it's our job to embrace it and make sure this is highlighted and never lost.

Author: Yup, gotta love those Oxford commas :slightly_smiling_face: 

The extra "and" also adds emphasis, I think.  I see now that I did the same thing for the `Inclusion and Diversity and Mutual Respect` section, but I'm pretty sure that was solely for emphasis and because I thought `Inclusion, Diversity, and Mutual Respect` seemed less genuine and more like ticking off a box.

The period after "important" is definitely a mistake, though.  @hobbs-Hobbler flagged it as well.

Let me know if people still want it changed, or if now that I've provided an explanation they have different thoughts.

> We believe that fun and enjoyment and wonder are important.and coexist with and enhance the other high bars that we set for the project.  Let's face it, if we aren't evoking these emotions in people then we're doing something wrong because what could be more awesome or inspiring than being able to create and shape a shared metaverse, from the three dimensional constructs to the underlying code!  Hubs presents us with infinite possibilities and we believe it's our job to embrace it and make sure this is highlighted and never lost.

Reviewer 2: Oh, I like the "no comma" when you meant fun.   I also can accept that you didn't want a box-ticking vibe. I'm OK with NO oxford comma in these cases.  Because....
![always be a pirate](https://github.com/user-attachments/assets/4fecc165-5a85-4e82-915b-3227bb0c3664)
